main.c

```diff
@@ -800,7 +792,19 @@ main
     cs_str_reset(cs, "debug_level", NULL);
   if (dfile)
     cs_str_reset(cs, "debug_file", NULL);
-
+  if (!dfile)
+  {
+    dfile = cs_subset_path(NeoMutt->sub, "debug_file");
+    cs_str_initial_set(cs, "debug_file", dfile, NULL);
+  }
```

Comment:
Ah!  We don't want this.
The config variables have default values, set in the code.
A special few of them have their defaults set at startup from environment variables or command line options,
e.g.

https://github.com/neomutt/neomutt/blob/34dc388b40666eb4d0db66596455330fc2664512/mutt_config.c#L242

Running `neomutt -l apple` will set the default, so that `reset debug_file` will revert to that command line value.

If the user has the config `set debug_file = "apple"` then `reset debug_file` will revert to `~/.neomuttdebug0`